### PR TITLE
Fix: nullability for managed object subclasses

### PR DESCRIPTION
**Pull request by KaterinaWire**
###  What's new in this PR?

Change nullability for managed object subclasses according to the data model definition.

### CODE REVIEW

File: Source/Model/Conversation/ZMConversation.m [hunk 1]
```diff
@@ -124,8 +124,8 @@ @interface ZMConversation ()
 @property (nonatomic) NSDate *lastReadServerTimeStamp;
 @property (nonatomic) NSDate *lastServerTimeStamp;
 @property (nonatomic) NSDate *clearedTimeStamp;
-@property (nonatomic) NSDate *archivedChangedTimestamp;
-@property (nonatomic) NSDate *silencedChangedTimestamp;
+@property (nonatomic, nullable) NSDate *archivedChangedTimestamp;
```
Comment (typfel) [inline Source/Model/Conversation/ZMConversation.m]: Nullability in generally specified in the header files. These two properties are already annotated in the `ZMConversation+Internal.h`. In Objective-C it's common to define properties twice, once in the public interface and again in the private interface.

File: Source/Model/Message/ZMImageMessage.m [hunk 2]
```diff
@@ -34,7 +34,7 @@
 @interface ZMImageMessage ()
 
 @property (nonatomic) BOOL isAnimatedGIF;
-@property (nonatomic) NSString *imageType;
+@property (nullable, nonatomic) NSString *imageType;
```
Comment (typfel) [inline Source/Model/Message/ZMImageMessage.m]: Already annotated in `ZMMessage+Internal.h.`

File: Source/Model/User/ZMUser.m [hunk 3]
```diff
@@ -140,12 +140,12 @@ + (BOOL)isTrackingLocalModifications
 
 @interface ZMUser ()
 
-@property (nonatomic) NSString *normalizedName;
+@property (nullable, nonatomic) NSString *normalizedName;
```
Comment (typfel) [inline Source/Model/User/ZMUser.m]: It's enough to annotate properties in the header file.

File: Source/Model/User/ZMUser.m [hunk 4]
```diff
@@ -140,12 +140,12 @@ + (BOOL)isTrackingLocalModifications
...
 @property (nonatomic, copy) NSString *name;
 @property (nonatomic) ZMAccentColor accentColorValue;
 @property (nonatomic, copy) NSString *emailAddress;
 @property (nonatomic, copy) NSString *phoneNumber;
-@property (nonatomic, copy) NSString *normalizedEmailAddress;
+@property (nullable, nonatomic, copy) NSString *normalizedEmailAddress;
```
Comment (typfel) [inline Source/Model/User/ZMUser.m]: It's enough to annotate properties in the header file.

File: Source/Model/Conversation/ZMConversation.m (hunk 1 above)
Comment (KaterinaWire) [reply Source/Model/Conversation/ZMConversation.m]: Fixed

File: Source/Model/Message/ZMImageMessage.m (hunk 2 above)
Comment (KaterinaWire) [reply Source/Model/Message/ZMImageMessage.m]: Fixed

File: Source/Model/User/ZMUser.m (hunk 3 above)
Comment (KaterinaWire) [reply Source/Model/User/ZMUser.m]: Fixed

File: Source/Model/User/ZMUser.m (hunk 4 above)
Comment (KaterinaWire) [reply Source/Model/User/ZMUser.m]: Fixed

**Comment by typfel** [comment]
The base class for all managed object entities is still missing nullability annotations `ZMMangedObject.h`.

**Comment by KaterinaWire** [comment]
> nullability annotations

@typfel Fixed

**Comment by KaterinaWire** [comment]
@zenkins release major

**Pull request merged**